tests/integration/kubernetes/k8s-guest-pull-image-signature.bats

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env bats
+# Copyright (c) 2024 IBM Corporation
+# Copyright (c) 2024 Alibaba Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/confidential_kbs.sh"
+
+export KBS="${KBS:-false}"
+
+setup() {
+    # TODO - do TEEs want to run these tests?
+    # if is_confidential_hardware; then
+    #     skip "Due to issues related to pull-image integration skip tests for ${KATA_HYPERVISOR}."
+    # fi
+
+    if ! is_confidential_runtime_class; then
+        skip "Test not supported for ${KATA_HYPERVISOR}."
+    fi
+
+    [ "${SNAPSHOTTER:-}" = "nydus" ] || skip "None snapshotter was found but this test requires one"
```

Comment:
I think this is going to skip the test for CRI-O - we're not using a snapshotter there.
Any way to verify that we're running with containerd before we check the snapshotter?

Reply:
That's a really good poin, but we have this same check in all of our guest-pull tests as we don't have any upstream testing involving cri-o at the moment, so it's effectively not supported. I think probably the best way to handle this is as part of cri-o enabling that @wainersm has been looking at?

We also need to find a way to remove the duplications and help make this sort of thing easier.

Reply:
Answering to myself: this is already the case for generic guest-pull feature, so I guess it is unrelated to this PR.

Reply:
Interesting, github have let me comment after you without showing me your comment :-/ 
I actually thought we have one crio test running (on garm?) but maybe it's not running those specific tests?

Reply:
Yeah, no confidential image test with cri-o yet

Reply:
Hi @littlejawa !

Yes, the crio test running on garm is testing only "vanilla" kata. As @stevenhorsman I plan to add a new job to test the CoCo enabled features with cri-o. The kata-deploy part is implemented already, it's really about creating a job and adapting the tests (as you realized, some made assumptions about containerd/nydus that doesn't apply for cri-o). I hope to work on that soon, I let you posted @littlejawa !